packages/core/echo/echo-schema/src/indexing/index-schema.ts

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright 2024 DXOS.org
+//
+import * as orama from '@orama/orama';
+
+import { Event } from '@dxos/async';
+import { invariant } from '@dxos/invariant';
+import { PublicKey } from '@dxos/keys';
+
+import {
+  type ObjectType,
+  type Index,
+  type IndexKind,
+  type IndexStaticProps,
+  type LoadParams,
+  staticImplements,
+} from './types';
+import { type Filter } from '../query';
+
+// TODO(mykola): Correct schema ref?
+type OramaSchemaType = orama.Orama<
+  {
+    system: {
+      type: { itemId: 'string' };
+    };
+  },
+  orama.IIndex<orama.components.index.Index>,
+  orama.IDocumentsStore<orama.components.documentsStore.DocumentsStore>
+>;
+
+@staticImplements<IndexStaticProps>()
+export class IndexSchema implements Index {
```

Comment:
Rename to type index, as it's indexing based on the object typename, not the schema contents